src/pygeoprocessing/geoprocessing.py

```diff
@@ -4065,6 +4078,108 @@ def _mult_op(base_array, base_nodata, scale, datatype):
     target_band = None
 
 
+def build_overviews(
+        raster_path, internal=False, resample_method='nearest',
+        overwrite=False, levels='auto'):
+    """Build overviews for a raster dataset.
+
+    Args:
+        raster_path (str): A path to a raster on disk and the band index to
+            build overviews for.
+        internal=False (bool): Whether to modify the raster when building
+            overviews. In GeoTiffs, this builds internal overviews when
+            ``internal=True``, and external overviews when ``internal=False``.
+        resample_method='nearest' (str): The resample method to use when
+            building overviews.
+        overwrite=False (bool): Whether to overwrite existing overviews, if
+            any exist.
+        levels='auto' (sequence): A sequence of integer overview levels. If
+            ``'auto'``, overview levels will be determined by using factors of
+            2 until the overview's x and y dimensions are both less than 256.
+
+    Example:
+        Generate overviews, regardless of whether overviews already exist
+        for the raster, letting the function determine the levels of overviews
+        to generate::
+
+            build_overviews(raster_path)
+
+        Generate overviews for 4 levels, at 1/2, 1/4, 1/8 and 1/16 the
+        resolution::
+
+            build_overviews(raster_path, levels=[2, 4, 8, 16])
+
+    Returns:
+        ``None``
+    """
+    # WarpOptions.this is None when an invalid option is passed, and it's a
```

Comment:
This ended up being kind of neat!  GDAL doesn't have something like a module-level mapping to get the human-readable types, and the mapping of `gdal.GRA_*` to human-readable names is [local to the `gdal.WarpOptions` parsing function.](https://github.com/OSGeo/gdal/blob/bfe1cfea364a0e185a792db67db7de2fc6aad902/swig/python/osgeo/gdal.py#L854).  I put together something in 7bdd1dbc8a4fc62015c0a737e4712b5022b75708 to get those human-readable names out of `gdal.WarpOptions` (avoiding `ast` inspection in case we can't read the source code like in pyinstaller binaries) and making them available at the module level.  This should make pygeoprocessing resilient to any changes to the set of resample algorithms GDAL offers, so long as they keep `gdal.WarpOptions` up to date!

454dda32c9251191f8f882b1b7a803d594e11693 takes this a step further and uses these human-readable resample algorithms in the docstrings where they are used.